embedded/sql/catalog.go

```diff
@@ -18,6 +18,8 @@ package sql
 type Catalog struct {
 	dbsByID   map[uint64]*Database
 	dbsByName map[string]*Database
+
+	mutated bool
```

Comment:
I was confused by the name of this member a bit, maybe something like `reloadNeeded` would be better here?

Reply:
I think it's better to label as dirty or mutated, reloading it's a decision, there may be another ways to solve the rollback than reloading. And this catalog implementation may require transactional support as well. But will be addressed in a separate work. This mutated was added in this PR but was actually missing.